Test/PublishPowerShellScript.Tests.ps1

```diff
@@ -0,0 +1,433 @@
+    {
+        $script:failed = $true
+        Write-Error -ErrorRecord $_ -ErrorAction $ErrorActionPreference
+    }
+}
+
+function Reset
+{
+    Write-DEbug 'PUBLISHPOWERSHELLSCRIPT  RESET'
+    Get-Module | Format-Table -AutoSize | Out-String | Write-Debug
+
+    $Global:Error | Format-List * -force | Out-String | Write-Debug
+
+    Reset-WhiskeyTestPSModule
+    Get-PSRepository | Where-Object 'Name' -Like 'Whiskey*' | Unregister-PSRepository
+    if( $script:repoToUnregister )
+    {
+        Unregister-PSRepository -Name $script:repoToUnregister
+    }
+}
+Reset
+
+function ThenFailed
+{
+    param(
+        [Parameter(Mandatory)]
+        $WithError
+    )
+
+    $script:failed | Should -BeTrue
+    Get-Error | Should -Match $WithError
+}
+
+function ThenRepository
+{
+    param(
+        [Parameter(Mandatory)]
+        [String] $Named,
+
+        [switch] $Exists,
+
+        [switch] $NotExists,
+
+        [switch] $NotRegistered,
+
+        [switch] $NotUnregistered
+    )
+
+    if( $NotRegistered )
+    {
+        Assert-MockCalled -CommandName 'Register-PSRepository' -ModuleName 'Whiskey' -Times 0
+    }
+
+    if( $Exists )
+    {
+        Get-PSRepository |
+            Where-Object 'Name' -EQ $Named |
+            Where-Object 'PublishLocation' -EQ $script:publishRoot |
+            Should -Not -BeNullOrEmpty
+    }
+
+    if( $NotExists )
+    {
+        Get-PSRepository |
+            Where-Object 'Name' -EQ $Named |
+            Should -BeNullOrEmpty
+    }
+
+    if( $NotUnregistered )
+    {
+        Assert-MockCalled -CommandName 'Unregister-PSRepository' -ModuleName 'Whiskey' -Times 0
+    }
+}
+
+function ThenScriptPublished
+{
+    param(
+        $To = $script:context.OutputDirectory.FullName,
+        $WithPrerelease = ''
+    )
+    
+    if( -not [IO.Path]::IsPathRooted($To) )
+    {
+        $To = Join-Path -Path $script:testRoot -ChildPath $To
+    }
+
+    Join-Path -Path $To -ChildPath "$($testScriptName).*.*.*$($WithPrerelease).nupkg" | Should -Exist
+}
+
+function ThenManifest
+{
+    param(
+        [String]$ManifestPath = (Join-Path -Path $testRoot -ChildPath "$($testScriptName)\$($testScriptName).ps1"),
+
+        [String]$HasPrerelease
+    )
+
+    Test-ScriptFileInfo -Path $ManifestPath | Should -Not -BeNullOrEmpty
+    Get-Content -Raw -Path $ManifestPath | Should -Match ".VERSION 1.2.3-$($HasPrerelease)"
+}
+
+function ThenSucceeded
+{
+    $script:failed | Should -BeFalse
+    Get-Error | Should -BeNullOrEmpty
+}
+
+function Get-Error
+{
+    [CmdletBinding()]
+    param(
+    )
+
+    # PackageManagement and PowerShellGet leave handled errors in Global:Error so we need to filter those errors out.
+    $Global:Error |
+        Where-Object 'ScriptStackTrace' -NotMatch '\bPowerShellGet\b' -ErrorAction Ignore |
+        Where-Object 'ScriptStackTrace' -NotMatch '\bPackageManagement\b' -ErrorAction Ignore
+}
+
+Describe 'PublishPowerShellScript.when publishing to repository that already exists' {

```

Comment:
Add a test that ensures `CredentialID` is used to get a credential that gets passed to `Register-PSRepository`.